app/services/xccdf_tailoring_file.rb

```diff
@@ -46,11 +46,30 @@ def profile_builder(xml)
   end
 
   def rule_selections_builder(xml)
-    @rule_ref_ids.each do |rule_ref_id, selected|
+    resolved_rules.each do |rule_ref_id, selected|
       xml[XCCDF].select(idref: rule_ref_id, selected: selected)
     end
   end
 
+  def resolved_rules
+    @rule_ref_ids.select { |_id, selected| selected }.each do |rule_ref_id, _selected|
+      rule_for(rule_ref_id)&.left_rule_group_relationships&.each do |rgr|
+        @rule_ref_ids[rule_ref_id] = false if conflict?(rgr)
```

Comment:
I think that the algorithm will be way more complex than this:
* where are you handling the requires?
* the text cited above mentions parent items deselection, but we don't keep track of parent items, so we need to populated that down into the siblings in the given group...

Anyway, this needs a bit more work.

Reply:
The requires is handled in the `conflict?` method but I guess that naming for that method is confusing. By "conflict" I just meant any conflict that would cause issues for the Rule like a requirement isn't satisfied but I could rename it

When it said parent item in the description, I thought that it meant the parent of the `<requires>` or `<conflicts>` element which would just be the Rule? Not the parent of the Rule with the `conflicts` or `requires` element. At least that's what it sounds like to me... Because it mentions to never update any other item anywhere else in the Benchmark besides the one that's currently being evaluated

Reply:
The parent item is the group I guess, if you look at the SCAP workbench, it will show the tree structure with the parent item.

Reply:
I put this in the description too but here's an example of how conflicts are resolved, from the xccdf specfications:
<img width="491" alt="Screen Shot 2022-04-26 at 2 45 59 PM" src="https://user-images.githubusercontent.com/87209745/165380347-ef490f7e-a7d0-4966-b0ce-be4fa023669b.png">

Step 4 shows the process of evaluating `Rule2` which requires `Rule3`. Since `Rule3` isn't selected, `Rule2` is unselected and this is how the conflict is resolved. 

I think that when it says `Also <xccdf:requires> and <xccdf:conflicts> elements SHALL only change their parent item;` means the parent of the requires or conflicts element so in this example `<xccdf:Rule id="xccdf_org.example_rule_Rule2" selected="true"> ` is the parent of `<xccdf:requires idref="xccdf_org.example_rule_Rule3"> `

Reply:
Also, I don't think Groups can technically be selected yet? Like it isn't possible to select a Group on the front end? So until we add that functionality, is evaluating only Rules for now good enough? 
Or I can update it to evaluate Groups in the conflict resolving algortihm too but for now, selected will always be false for every Group.

Reply:
/retest